python/tests/test_sbml_import.py

```diff
@@ -722,3 +722,50 @@ def test_hardcode_parameters(simple_sbml_model):
             constant_parameters=["p1"],
             hardcode_symbols=["p1"],
         )
+
+
+def test_constraints():
+    """Test non-negativity constraint handling."""
+    from amici.antimony_import import antimony2amici
+
+    ant_model = """
+    model test_non_negative_species
+        species A = 10
+        species B = 0
+        # R1: A => B; k1f * sqrt(A)
+        R1: A => B; k1f * max(0, A)
+        k1f = 1e10
+    end
+    """
+    module_name = "test_non_negative_species"
+    with TemporaryDirectory(prefix=module_name) as outdir:
+        antimony2amici(
+            ant_model,
+            model_name=module_name,
+            output_dir=outdir,
+            compute_conservation_laws=False,
+        )
+        model_module = amici.import_model_module(
+            module_name=module_name, module_path=outdir
+        )
+        amici_model = model_module.getModel()
+        amici_model.setTimepoints(np.linspace(0, 100, 200))
+        amici_solver = amici_model.getSolver()
+        rdata = amici.runAmiciSimulation(amici_model, amici_solver)
+        assert rdata.status == amici.AMICI_SUCCESS
+        # should be non-negative in theory, but is expected to become negative
+        #  in practice
+        assert np.any(rdata.x < 0)
+
+        amici_solver.setRelativeTolerance(1e-14)
+        amici_solver.setConstraints([1.0, 1.0])
```

Comment:
can we add an enum for `0.0`/`1.0`/`2.0`/`-1.0`/`-2.0` values for a bit more user convenience??

Reply:
also, probably makes sense to automatically add constraints when using `Model::setStateIsNonNegative` and add this behaviour to the documentation of setConstraints and setStateIsNonNegative.

Reply:
Added a python enum for now.

Reply:
> can we add an enum for `0.0`/`1.0`/`2.0`/`-1.0`/`-2.0` values for a bit more user convenience??

done

> also, probably makes sense to automatically add constraints when using `Model::setStateIsNonNegative` and add this behaviour to the documentation of setConstraints and setStateIsNonNegative.

I am not completely sure about this one. While I agree that it makes sense to imply the constraints in `setStateIsNonNegative` `True`, this would lead to some funny behavior when disabling non-negativity. 

```
# I want to enable constraints:
model.setConstraints([non_negative])
# I want to enable the max(0, state) in addition:
model.setStateIsNonNegative([True])
# I decide to disable the latter
model.setStateIsNonNegative([False]) # implies setConstraints([none]) ?!
# now it also cleared my original constraint
```

I think I'd prefer to keep them independent for now.